src/metax_api/models/catalog_record_v2.py

```diff
@@ -103,6 +102,12 @@ def _pre_create_operations(self):
 
         self._set_api_version()
 
+        if self._save_as_draft():
+            self.state = self.STATE_DRAFT
+            self.research_dataset['preferred_identifier'] = 'draft:%s' % self.identifier
+            if self._get_preferred_identifier_type_from_request() == IdentifierType.DOI:
+                self.use_doi_for_draft = True
```

Comment:
These all could be handled in the post_create_operations to avoid creating the preferred identifier twice and avoiding code scattering aswell

Reply:
Good point, modified now